Add onelogin and onelogin-aws-assume-role python libraries

gjhenrique opened the pull request:
My company uses [onelogin](https://www.onelogin.com/) to store the AWS accounts and roles. We have an internal tool that wraps the python-based [aws-login-aws-assume-role](https://github.com/onelogin/onelogin-python-aws-assume-role)

The plan is to add this package to nixpkgs together with its dependency `onelogin` (version bigger than 3) and `onelogin_2`. This tool is only compatible with 2.x.x, so `onelogin` wasn't needed, but it's a nice to have.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/2.nix:
```diff
@@ -0,0 +1,50 @@
+    description = "OpenAPI Specification for OneLogin";
+    homepage = "https://github.com/onelogin/onelogin-python-sdk";
+    license = lib.licenses.mit;
+    maintainers = [ ];
```
same as above

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/default.nix:
```diff
@@ -0,0 +1,55 @@
+    description = "OpenAPI Specification for OneLogin";
+    homepage = "https://github.com/onelogin/onelogin-python-sdk";
+    license = lib.licenses.mit;
+    maintainers = [ ];
```
same as above

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/default.nix:
```diff
@@ -0,0 +1,55 @@
+    urllib3
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    python -m unittest
+    runHook postCheck
+  '';
```
Please use pytestCheckHook instead

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/default.nix:
```diff
@@ -0,0 +1,55 @@
+    hash = "sha256-4REMLFMVDTEu0BpxhlF8nfXGLW8/UwoejXVT4P7jTuo=";
+  };
+
+  sourceRoot = "${src.name}";
```
same question as above

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/default.nix:
```diff
@@ -0,0 +1,55 @@
+  src = fetchFromGitHub {
+    owner = "onelogin";
+    repo = "onelogin-python-sdk";
+    rev = version;
```
```suggestion
    tag = version;
```

SuperSandro2000 reviewed (commented):
onelogin seems to be unmaintained since 2 years. For what do we need those libraries?

gjhenrique commented:
> onelogin seems to be unmaintained since 2 years. For what do we need those libraries?

My company uses OneLogin as the single sign-on for services, including authenticating with AWS via SAML

This python library is their recommendation to get new short-lived AWS credentials. Even though it looks unmaintained, it's the only use AWS in the cli

gjhenrique commented on pkgs/development/python-modules/onelogin/2.nix:
```diff
@@ -0,0 +1,50 @@
+    hash = "sha256-KfrYvsNr+RZOTmhJHYcLhxvKy00vUmejvcX9d1e6Rqo=";
+  };
+
+  sourceRoot = "${src.name}";
```
Not really. More like copy and pasting, but I will remove it

gjhenrique committed:
maintainers: add gjhenrique

gjhenrique commented:
Hi @SuperSandro2000 thanks for the review :smile:. PTAL

SuperSandro2000 commented on pkgs/development/python-modules/onelogin/default.nix:
```diff
@@ -0,0 +1,53 @@
+    urllib3
+  ];
+
+  pytestCheckHook = ''
+    runHook preCheck
+    python -m unittest
+    runHook postCheck
+  '';
```
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```

gjhenrique replied:
Done

SuperSandro2000 reviewed (commented):
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

gjhenrique committed:
python312Packages.onelogin: init at 3.1.15

gjhenrique committed:
python312Packages.onelogin_2: init at 2.0.4

gjhenrique committed:
python312Packages.onelogin-aws-assume-role: init at 1.10.1

gjhenrique commented:
Got it. I somehow missed the naming commit part :sweat_smile:

1. I renamed the commits for each package to have its own commit

2. Using ``rev` resulted in the error, so I reverted the code back to `tag`

```
       error: function 'anonymous lambda' called without required argument 'rev'
       at /nix/store/idq1bmwwy9rlkc21hccvx42wlwpxsx1f-source/pkgs/build-support/fetchgithub/default.nix:4:1:
            3| lib.makeOverridable (
            4| { owner, repo, rev, name ? "source"
             | ^
            5| , fetchSubmodules ? false, leaveDotGit ? null

```
